### PR TITLE
CTCTRADERS-2592 Add Limits

### PR DESCRIPTION
**Pull request by carrielies**

### CODE REVIEW

File: app/controllers/addItems/containers/AddAnotherContainerController.scala [hunk 1]
```diff
@@ -46,22 +47,25 @@ class AddAnotherContainerController @Inject() (
...
+  config: FrontendAppConfig
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport
     with NunjucksSupport {
 
-  private val form     = formProvider()
   private val template = "addItems/containers/addAnotherContainer.njk"
 
+  def allowMoreContainers(ua: UserAnswers, itemIndex: Index): Boolean =
```
Comment (jackheslop96) [inline app/controllers/addItems/containers/AddAnotherContainerController.scala:59]: Can be made private I'm guessing?

File: app/controllers/addItems/documents/AddAnotherDocumentController.scala [hunk 2]
```diff
@@ -50,19 +49,23 @@ class AddAnotherDocumentController @Inject() (
...
   val controllerComponents: MessagesControllerComponents,
-  renderer: Renderer
+  renderer: Renderer,
+  config: FrontendAppConfig
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport
     with NunjucksSupport {
 
+  def allowMoreDocuments(ua: UserAnswers, index: Index): Boolean =
```
Comment (jackheslop96) [inline app/controllers/addItems/documents/AddAnotherDocumentController.scala:59]: Can be made private?

File: app/controllers/addItems/packagesInformation/AddAnotherPackageController.scala [hunk 3]
```diff
@@ -43,13 +44,15 @@ class AddAnotherPackageController @Inject() (
...
   val controllerComponents: MessagesControllerComponents,
+  config: FrontendAppConfig,
   renderer: Renderer
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport
     with NunjucksSupport {
 
-  private val form = formProvider()
+  def allowMorePackages(ua: UserAnswers, itemIndex: Index): Boolean =
```
Comment (jackheslop96) [inline app/controllers/addItems/packagesInformation/AddAnotherPackageController.scala:54]: Same as above

File: conf/messages.en [hunk 4]
```diff
@@ -637,6 +641,7 @@ addAnotherPackage.yesNo = Do you need to add another package?
 addAnotherPackage.error.required = Select yes if you need to add another package
 addAnotherPackage.change.hidden = Change {0}
 addAnotherPackage.packageList.label = Package {0}
+addAnotherPackage.packageList.label = Package {0}
```
Comment (jackheslop96) [inline conf/messages.en:644]: Duplicate?

File: conf/messages.en [hunk 5]
```diff
@@ -833,6 +839,7 @@ addAnotherSpecialMention.heading.singular = You have added 1 special mention for
 addAnotherSpecialMention.heading.plural = You have added {0} special mentions for item {1}
 addAnotherSpecialMention.addAnother = Do you need to add another special mention?
 addAnotherSpecialMention.specialMentionList.label = Special mention {0}
+addAnotherSpecialMention.maxLimit.label = You cannot add any more special mentions
```
Comment (jackheslop96) [inline conf/messages.en:842]: No full stop - guessing these should all be conistsnet?

File: conf/messages.en [hunk 6]
```diff
@@ -1104,6 +1111,7 @@ addAnotherCountryOfRouting.heading.plural = You have added {0} countries of rout
 addAnotherCountryOfRouting.h2 = Do you need to add another country of routing?
 addAnotherCountryOfRouting.error.required = Select yes if you need to add another country of routing
 addAnotherCountryOfRouting.countryOfRoutingList.label = Country of routing {0}
+addAnotherCountryOfRouting.maxLimit.label = You cannot add any more country of routing
```
Comment (jackheslop96) [inline conf/messages.en:1114]: Should these be `**countries** of routing`?

File: conf/messages.en (hunk 5 above)
Comment (carrielies) [reply conf/messages.en:842]: I'll take the full stops off the others

Comment (carrielies) [reply conf/messages.en:842]: There's more with the full stop, so I'll add the full stop to all

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**